src/witness_generator/machines/fixed_lookup_machine.rs

```diff
@@ -76,60 +74,82 @@ impl FixedLookup {
         left: &[Result<AffineExpression, EvalError>],
         right: &SelectedExpressions,
     ) -> EvalResult {
-        let left_key = match left[0].clone() {
-            Ok(v) => match v.constant_value() {
-                Some(v) => Ok(v),
-                None => Err(format!(
-                    "First expression needs to be constant but is not: {}.",
-                    v.format(fixed_data)
-                )),
-            },
-            Err(err) => Err(format!("First expression on the LHS is unknown: {err}")),
-        }?;
-
-        let right_key = right.expressions.first().unwrap();
-        let rhs_row = if let Expression::PolynomialReference(poly) = right_key {
-            // TODO we really need a search index on this.
-            fixed_data.fixed_cols
-                .get(poly.name.as_str())
-                .and_then(|values| values.iter().position(|v| *v == left_key))
-                .ok_or_else(|| {
-                    format!(
-                        "Unable to find matching row on the RHS where the first element is {left_key} - only fixed columns supported there."
-                    )
-                })
-                .map(|i| i as DegreeType)
-        } else {
-            Err("First item on the RHS must be a polynomial reference.".to_string())
-        }?;
+        let columns = right
+            .expressions
+            .iter()
+            .map(|right_key| {
+                if let Expression::PolynomialReference(poly) = right_key {
+                    fixed_data
+                        .fixed_cols
+                        .get(poly.name.as_str())
+                        .ok_or_else(|| {
+                            format!(
+                                "Unable to find a fixed column for the fixed lookup: {}",
+                                poly.name
+                            )
+                        })
+                } else {
+                    Err("RHS must be a polynomial reference.".to_string())
```

Comment:
We should rule this out already in the main plookup function, where we can return `None` saying that this machine cannot handle that lookup. There is a function called `is_simple_poly` that you can move to utils and use.